src/functions-and-arrays.js

```diff
@@ -1,41 +1,141 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
+function maxOfTwoNumbers(num1, num2) {
+
+  if(num1>num2) {
+    return num1
+  }
+ 
+else {
+  return num2
+}
+}
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(names) {
+let bigName =""
 
+if (names.length === 0) {
+  return null
+}
+
+for(let name of names) {
+  if( name.length > bigName.length) {
+ bigName = name
+  }
+}
+return bigName
+}
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(numArr) {
+  let sum = 0
+
+if(numArr.length===0){
+  sum=0
+}
+
+  for(let num of numArr){
+
+     sum+=num
+   }
+   return sum
+  }
 
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+function sum(sumArray) {
+
+  let sumB=0
+
+  for (let numSum of sumArray) {
+    
+    switch ( typeof numSum ) {
+      case 'number' :
+        sumB+=numSum
+        break
+      case 'string' :
+        sumB+=numSum.length
+        break
+
+        case 'boolean':
```

Comment:
great use of switch !!

remember javascript will turn a boolean into a number if you try to sum it up so you could do

`sumB += numSum`